src/mlpack/methods/nca/nca_softmax_error_function.hpp

```diff
@@ -79,6 +79,22 @@ class SoftmaxErrorFunction
   double Evaluate(const arma::mat& covariance, const size_t i);
 
   /**
+   * Evaluate the softmax objective function for the given covariance matrix on
+   * the given batch size from a given inital point of the dataset.
+   * This is the separable implementation, where the objective 
+   * function is decomposed into the sum of many objective
+   * functions, and here, only one of those constituent objective functions is
+   * returned.
+   *
+   * @param covariance Covariance matrix of Mahalanobis distance.
+   * @param begin Index of the initial point to use for objective function.
+   * @param batchSize Number of points to use for objective function.
+   */
+  double Evaluate(const arma::mat& covariance,
+                  const size_t begin,
+                  const size_t batchSize) const;
```

Comment:
hm, if I remove the const from the Evaluate and Gradient function, it "works" for me, I end up with another error about GradType but that should be easy to solve.